contracts/Registry.sol

```diff
@@ -6,17 +6,19 @@ contract Registry {
     using strings for *;
 
     // todo: choose what to index
-    event LogPublish(address issuer, address subject, bytes32 action, bytes32 contentType, string cid);
+    event LogPublish(address issuer, address subject, bytes32 action, bytes32 contentType, string cid, uint prevblock);
     event LogPublishString(address issuer, string message);
 
+    uint currBlock = block.number;
     // recognized actions == CRUD.
     function publish(address[] _subjects, bytes32[] _actions, bytes32[] _contentTypes, string _cids) public {
         require(_subjects.length == _actions.length && _actions.length == _contentTypes.length);
         var prepCids = _cids.toSlice();
         for (uint8 i = 0; i < _contentTypes.length; i++) {
             string memory cid = prepCids.split("|".toSlice()).toString();
             require(bytes(cid).length != 0);
-            LogPublish(msg.sender, _subjects[i], _actions[i], _contentTypes[i], cid);
+            LogPublish(msg.sender, _subjects[i], _actions[i], _contentTypes[i], cid, currBlock);
+            currBlock = block.number;
```

Comment:
I'm wondering if this shouldn't be after the for loop? eg, all in this block should point to the previous block vs only the first one pointing the new block? What do you think?

I think the former makes more sense (all in one block) since that's you would usually care about? vs fetching one log at a time? Lemme ask Pelle to confirm as well.

Reply:
uPort doesn't have a for loop, but it looks they are just changing like you did. Let's keep it at that for now [like you did it].

There's is one thing though. It's storing it every single time when it's updated. Can you add a check to see if it hasn't already been stored (to save gas)?